File: Samples/UIKit/Objective-C/ViewController.m
//
//  ViewController.m
//  UIKit Sample
//
//  Created by Сергій Попов on 27.10.2022.
//

#import "ViewController.h"

@import Setapp;

@interface ViewController () <STPManagerDelegate>

@property (weak, nonatomic) IBOutlet UIImageView *statusImage;
@property (weak, nonatomic) IBOutlet UILabel *statusLabel;
@property (weak, nonatomic) IBOutlet UILabel *infoLabel;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self updateInfoLabel];
    [self updateWithSubscription:STPManager.sharedInstance.subscription];
    
    STPManager.sharedInstance.delegate = self;
}

#pragma mark Setapp

- (void)setappManager:(STPManager *)manager didUpdateSubscriptionTo:(STPSubscription *)newSetappSubscription {
    [self updateWithSubscription:newSetappSubscription];
}

#pragma mark Helpers

- (void)updateWithSubscription:(STPSubscription *)subscription {
    if (subscription) {
        self.statusImage.image = [UIImage systemImageNamed:subscription.isActive ? @"checkmark.circle" : @"x.circle"];
        self.statusImage.tintColor = subscription.isActive ? UIColor.systemGreenColor : UIColor.systemRedColor;
        self.statusLabel.text = [NSString stringWithFormat:@"%@ Setapp Subscription", subscription.isActive ? @"Active" : @"Inactive"];
    }
    else {
        self.statusImage.image = [UIImage systemImageNamed:@"qrcode"];
        self.statusImage.tintColor = UIColor.tintColor;
        self.statusLabel.text = @"Activate via QR Code";
    }
}

- (void)updateInfoLabel {
    self.infoLabel.text = NSBundle.mainBundle.infoDictionary[@"CFBundleName"];
}

@end

File: Samples/UIKit/Swift/ViewController.swift
//
//  ViewController.swift
//  UIKit Sample
//
//  Created by Сергій Попов on 27.10.2022.
//

import Setapp
import UIKit

class ViewController: UIViewController {
    @IBOutlet var statusImage: UIImageView!
    @IBOutlet var statusLabel: UILabel!
    @IBOutlet var infoLabel: UILabel!

    override func viewDidLoad() {
        super.viewDidLoad()

        let targetName = Bundle.main.infoDictionary?["CFBundleName"] as? String
        infoLabel.text = targetName

        update(with: SetappManager.shared.subscription)
        SetappManager.shared.delegate = self
    }
}

// MARK: - SetappManagerDelegate

extension ViewController: SetappManagerDelegate {
    func setappManager(_: SetappManager, didUpdateSubscriptionTo newSetappSubscription: SetappSubscription) {
        update(with: newSetappSubscription)
    }
}

// MARK: Helpers

extension ViewController {
    private func update(with subscription: SetappSubscription?) {
        if let subscription {
            statusImage.image = UIImage(systemName: subscription.isActive ? "checkmark.circle" : "x.circle")
            statusImage.tintColor = subscription.isActive ? .systemGreen : .systemRed
            statusLabel.text = "\(subscription.isActive ? "Active" : "Inactive") Setapp Subscription"
        } else {
            statusImage.image = UIImage(systemName: "qrcode")
            statusImage.tintColor = .tintColor
            statusLabel.text = "Activate via QR Code"
        }
    }

    private func updateAppInfo() {
        let targetName = Bundle.main.infoDictionary?["CFBundleName"] as? String
        infoLabel.text = targetName
    }
}
